### ISSUE TITLE
typing info not available to extensions by default

### ISSUE THREAD

**Issue by boneskull** [appium/appium#17141]
I noticed that if an extension has Appium as a dev dep (as it should), the typing information is incomplete.  Specifically, `@appium/types` is not also a dev dep of the extension--and even if it is--the definition of `Driver` is incomplete due to a missing `type-fest` dependency. 

In practice, something like this does not elicit type information about `BaseDriver`:

```js
import {BaseDriver} from 'appium/driver';

export class MyDriver extends BaseDriver {
  // ...
}
```

Probably makes sense to add `@appium/types` as a prod dep of `appium`, and a few external typing modules as prod deps of `@appium/types` (`type-fest`, `@types/express`, `@types/npmlog`, etc.).

**Comment by boneskull** [comment]
(That change happened in f6cbfd77eb4e1e7ff722782756becb4a8e0113e5)

**Comment by boneskull** [comment]
Changing the links back to relative breaks smoke tests 🤦

**Comment by boneskull** [comment]
I think we can leave them as they are. but npm 6 won't be supported for a dev env.

**Comment by boneskull** [comment]
closed via f384d5360d446b30211757d5d5552c534071e14e
